File: kubernetes/minikube/plugins/js/traceheader-js/handler.js
const { v4: uuidv4 } = require("uuid");

const TraceHeader = {
  PRIORITY: 1000,
  VERSION: "1.0.0",
};

TraceHeader.header_filter = function (conf) {
  const trace_id = uuidv4();

  kong.response.setHeader(conf.trace_header, trace_id);
};

module.exports = TraceHeader;

File: kubernetes/minikube/plugins/lua/traceheader-lua/handler.lua
local uuid = require "resty.uuid"
local TraceHeader = {}

TraceHeader.PRIORITY = 1000
TraceHeader.VERSION = "1.0.0"

function TraceHeader:header_filter(conf)
    local trace_id = uuid.generate()

    kong.response.set_header(conf.trace_header, trace_id)
end

return TraceHeader
